sql/scan.go

```diff
@@ -764,6 +764,16 @@ func (v *qnameVisitor) Visit(expr parser.Expr, pre bool) (parser.Visitor, parser
 	}
 
 	switch t := expr.(type) {
+	case *qvalue:
+		// We will encounter a qvalue in the expression during retry of an
+		// auto-transaction. When that happens, we've already gone through
+		// qualified name normalization and lookup, we just need to hook the qvalue
+		// up to the scanNode.
+		//
+		// TODO(pmattis): Should we be more careful about ensuring that the various
+		// statement implementations do not modify the AST nodes they are passed?
+		return v, v.getQVal(t.col)
```

Comment:
Yes, we're taking work for one loop iteration and using it in the next. This isn't intentional, just an artifact of the way the code was written. As the `TODO` mentions, it would probably be better for this not to happen.